service/api_task.go

```diff
@@ -236,10 +236,18 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 
 	if checkHostHealth(currentHost) {
 		// set the needs new agent flag on the host
-		if err := currentHost.SetNeedsNewAgent(true); err != nil {
-			grip.Error(message.WrapErrorf(err, "error indicating host %s needs new agent", currentHost.Id))
-			gimlet.WriteResponse(w, gimlet.MakeJSONInternalErrorResponder(err))
-			return
+		if currentHost.Distro.LegacyBootstrap() {
+			if err := currentHost.SetNeedsNewAgent(true); err != nil {
+				grip.Error(message.WrapErrorf(err, "error indicating host %s needs new agent", currentHost.Id))
+				gimlet.WriteResponse(w, gimlet.MakeJSONInternalErrorResponder(err))
+				return
+			}
+		} else {
+			if err := currentHost.SetNeedsNewAgentMonitor(true); err != nil {
+				grip.Error(message.WrapErrorf(err, "error indicating host %s needs new agent monitor", currentHost.Id))
+				gimlet.WriteResponse(w, gimlet.MakeJSONInternalErrorResponder(err))
+				return
+			}
```

Comment:
can we push this logic down out of the service package. Seems like we there's an obvious "DoBootstrap" refactoring, so that if we were to add a new bootstrap method, we wouldn't have to go hunt all these things down. 

I'd embark on this refactoring by making the "LegacyBootstrap" method private and/or a property of the host type (the distro type is confusing.... it's a property of the host and not the current distro configuration.)

Reply:
I moved the legacy/non-legacy handling into `(*Host).SetNeedsAgentDeploy()`.